Gem/Code/Source/GameState/GameStateMatchInProgress.cpp

```diff
@@ -26,9 +32,26 @@ namespace MultiplayerSample
 
     void GameStateMatchInProgress::OnExit()
     {
+        PlayerCoinCollectorNotificationBus::Handler::BusDisconnect();
         m_roundChangedHandler.Disconnect();
     }
 
+    void GameStateMatchInProgress::OnPlayerCollectedCoinCountChanged([[maybe_unused]] Multiplayer::NetEntityId playerEntity,
+        uint16_t coinsCollected)
+    {
+        if (coinsCollected >= m_winningCoinCount)
+        {
+            const auto state = GameState::GameStateRequests::CreateNewOverridableGameStateOfType<GameStateMatchEnded>();
+            GameState::GameStateRequestBus::Broadcast(&GameState::GameStateRequestBus::Events::ReplaceActiveGameState, state);
+        }
+    }
+
+    int GameStateMatchInProgress::GetNotificationOrder()
+    {
+        // Putting this state as the last handler so that the player states are updated, otherwise the winning count might be off.
+        return LastNotificationOrder;
```

Comment:
feels odd to have a generally named value in the larger `MultiplayerSample` namespace living in the player coin collection bus (which is where I think this is coming from?). I'm guessing 1000 would work as a general "last notification order", but is there somewhere more , well, general to keep it?

Reply:
I haven't see a central place. We could put it into `MultiplayerSampleTypes.h` but I think these values are ebus specific to each ebus as is done TickBus: `Code\Framework\AzCore\AzCore\Component\TickBus.h`